src/tools/networkTools.cpp

```diff
@@ -211,39 +211,39 @@ std::map<std::string, std::string> getNetworkInterfaces() {
   return result;
 }
 
-
-std::string getBestPublicIp(bool ipv6) {
-  IpAddress bestPublicIp = IpAddress{"127.0.0.1","::1"};
+IpAddress getBestPublicIps(IpMode mode) {
+  IpAddress bestPublicIps = IpAddress{"127.0.0.1", "::1"};
   std::map<std::string, IpAddress> interfaces = getNetworkInterfacesIPv4Or6();
-
+  bool validMode4 = (mode == IpMode::ipv4 || mode == IpMode::all);  // Mode is valid to support ipv4
+  bool validMode6 = (mode == IpMode::ipv6 || mode == IpMode::all);  // Mode is valid to support ipv6
 #ifndef _WIN32
   const char* const prioritizedNames[] = { "eth0", "eth1", "wlan0", "wlan1", "en0", "en1" };
-  for(auto name: prioritizedNames) {
-    auto it=interfaces.find(name);
-    if(it != interfaces.end() && !(ipv6 && (*it).second.addr6.empty())) {
-      bestPublicIp = (*it).second;
-      break;
-    }
+  for(const auto& name: prioritizedNames) {
+    const auto it = interfaces.find(name);
+    if(it == interfaces.end()) continue;
+    const IpAddress& interfaceIps = it->second;
+    if(!bestPublicIps.addr.empty() && validMode4 && !interfaceIps.addr.empty()) 
+      bestPublicIps.addr = interfaceIps.addr;
+    if(!bestPublicIps.addr6.empty() && validMode6 && !interfaceIps.addr6.empty()) 
+      bestPublicIps.addr6 = interfaceIps.addr6;
   }
 #endif
-
   const char* const prefixes[] = { "192.168", "172.16.", "10.0" };
-  for(auto prefix : prefixes){
-    for(auto& itr : interfaces) {
-      std::string interfaceIp(itr.second.addr);
-      if (interfaceIp.find(prefix) == 0 && !(ipv6 && itr.second.addr6.empty())) {
-        bestPublicIp = itr.second;
-        break;
-      }
+  for (const auto& prefix : prefixes) {
+    for (const auto& [_, interfaceIps] : interfaces) {
+        if (!bestPublicIps.addr.empty() && validMode4 && !interfaceIps.addr.empty() && interfaceIps.addr.find(prefix) == 0)
+            bestPublicIps.addr = interfaceIps.addr;
+        if (!bestPublicIps.addr6.empty() && validMode6 && !interfaceIps.addr6.empty())
```

Comment:
Is the `interfaceIps.addr.find(prefix) == 0` check omitted intentionally?

Reply:
Yes, the reason it is omitted is because we only provide v4 prefixes.

Reply:
Would it be beneficial to provide some?

Reply:
> Yes, the reason it is omitted is because we only provide v4 prefixes.

The purpose of this piece of code is to select an IP address from a local network, though I am not sure why the `169.254` prefix (see https://en.wikipedia.org/wiki/Link-local_address) has not been included. Your selection of an IPv6 address fails to meet that criteria. Why don't you filter based on the same IPv4 check (`interfaceIps.addr.find(prefix) == 0`) but use the IPv6 address (`interfaceIps.addr6`)?

> Would it be beneficial to provide some for v6?

According to https://en.wikipedia.org/wiki/Link-local_address#IPv6

* In [IPv6](https://en.wikipedia.org/wiki/IPv6), unicast link-local addresses are assigned from the block *`fe80::/10`*

however, unfortunately, it seems like `fe80` cannot be used as a prefix because only its higher 10 bits matter. If we decide to support it a slightly more complex check will be required. But let's not go after it now.

Reply:
Ok so if I understand correctly, for this point, I should just add `169.254` for the v4 prefixes, right?

Reply:
No, the main concern was about the application of the prefix check when selecting the IPv6 address. The exact instructions what should be done is provided in the new review.

Reply:
> Ok so if I understand correctly, for this point, I should just add `169.254` for the v4 prefixes, right?

Adding the `169.254` prefix may have been a mistake. While testing the fix for #1144 I ran against a situation when the public API displayed by `kiwix-serve` starts with this prefix but that address doesn't work.